cmd/clusterctl/internal/util/objs.go

```diff
@@ -96,3 +97,48 @@ func IsSharedResource(o unstructured.Unstructured) bool {
 	}
 	return false
 }
+
+// FixImages alters images using the give alter func
+// NB. The implemented approach is specific for the provider components YAML & for the cert-manager manifest; it is not
+// intended to cover all the possible objects used to deploy containers existing in Kubernetes.
+func FixImages(objs []unstructured.Unstructured, alterImageFunc func(image string) (string, error)) ([]unstructured.Unstructured, error) {
+	// look for resources of kind Deployment and alter the image
+	for i := range objs {
+		o := &objs[i]
+		if o.GetKind() == deploymentKind {
```

Comment:
i would have:
```
if o.GetKind() != deploymentKind {
	continue
}
```
early here to avoid indenting the larger branch.

Reply:
done!